### PR TITLE
Feature/create event page

### PR DESCRIPTION
**Pull request by gor017**

### CODE REVIEW

File: src/assets/style/globalVariables.ts [hunk 1]
```diff
@@ -30,33 +30,51 @@ export const color = {
   closeButton: "#4F4F4F",
   overViewBoxshadow: "rgba(0, 0, 0, 0.15)",
   overViewBoxColor: "#1F6F8B",
+  tableBorderColor: "#BDBDBD",
+  redColor: "red",
```
Comment (ArthurO2002) [inline src/assets/style/globalVariables.ts:34]: use HEX colors everywhere

File: src/assets/styleJs/events/createEvent.ts [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+    alignItems: "center",
+    width: "40%",
+    marginBottom: margin.marginBottom25
+  },
+  eventInfoBoxEvaluatee: {
+    display: "flex",
+    alignItems: "center",
+    width: "40%",
+    marginBottom: margin.marginBottom25,
+  },
```
Comment (ArthurO2002) [inline src/assets/styleJs/events/createEvent.ts:26]: same thing in 2 places use the same class for both

File: src/assets/styleJs/events/createEvent.ts [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+    justifyContent: "space-evenly",
+    flexDirection: "column",
+    height:'100%'
+  },
+  evaluateeCardInputBox: {
+    display: "flex",
+    justifyContent: "space-evenly",
+    flexDirection: "column",
+    height:'100%'
+  },
```
Comment (ArthurO2002) [inline src/assets/styleJs/events/createEvent.ts:54]: do not write same thing for classes just use the same class

File: src/assets/styleJs/events/createEvent.ts [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+    padding: padding.padding0
+  },
+  evaluatorHeaderName: {
+    fontWeight: font.fontWeight500,
+    fontSize: font.fontSize16
+  },
+  evaluateeHeaderName: {
+    fontWeight: font.fontWeight500,
+    fontSize: font.fontSize16
+  },
```
Comment (ArthurO2002) [inline src/assets/styleJs/events/createEvent.ts:84]: Here too

File: src/assets/styleJs/events/createEvent.ts [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+  stepperStack: {
+    width: '70%',
+    padding: padding.padding0,
+  },
+  stepperRoot: {
+    padding: padding.padding0
+  },
+  stepperStep: {
+    padding: padding.padding0
+  }
```
Comment (ArthurO2002) [inline src/assets/styleJs/events/createEvent.ts:221]: same thing for both

do not use 0 as variable

File: src/pages/events/createEvents/EventsContext.tsx [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  }
+}
+
+
+export const ContextProvider = ({children}: EventProviderProps) => {
+  const [state, dispatch] = useReducer(eventReducer, initialState)
+  const value = React.useMemo(() => ({
+    state, dispatch
+  }), []);
+
```
Comment (ArthurO2002) [inline src/pages/events/createEvents/EventsContext.tsx:69]: is useMemo nesesary here?

File: src/pages/events/createEvents/stepper/ColorlibStepIconRoot.tsx [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import {styled} from "@mui/material/styles";
+
+const ColorlibStepIconRoot = styled('div')<{
+  ownerState: { completed?: boolean; active?: boolean };
+}>(({ ownerState }) => ({
+  backgroundColor:  '',
+  border:'2px solid #00A3FF',
```
Comment (ArthurO2002) [inline src/pages/events/createEvents/stepper/ColorlibStepIconRoot.tsx:7]: use variables

File: src/pages/events/createEvents/stepper/ColorlibStepIconRoot.tsx [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  display: 'flex',
+  borderRadius: '50%',
+  justifyContent: 'center',
+  alignItems: 'center',
+  ...(ownerState.active && {
+    backgroundColor:'transparent',
+  }),
+  ...(ownerState.completed && {
+    backgroundColor:'#00A3FF',
+    color:'#fff'
```
Comment (ArthurO2002) [inline src/pages/events/createEvents/stepper/ColorlibStepIconRoot.tsx:21]: variables for colors and values

File: src/pages/events/createEventsPageComponenets/CreateEvetsStepper.tsx [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+import useCreateEventStyles from "../../../assets/styleJs/events/createEvent";
+import ColorlibConnector from "../createEvents/stepper/ColorlibConnector";
+
+const ColorlibStepIcon=({active, className, completed, icon}: StepIconProps) =>{
+
+  const icons: { [index: string]: string } = {
+    1: '1',
+    2: '2',
+    3: '3',
+  };
```
Comment (ArthurO2002) [inline src/pages/events/createEventsPageComponenets/CreateEvetsStepper.tsx:17]: ?

File: src/pages/events/createEventsPageComponenets/CreateEvetsStepper.tsx [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+
+  return (
+    <Stack className={classes.stepperStack}>
+
+      <Stepper activeStep={1} connector={<ColorlibConnector/>} className={classes.stepperRoot}>
+        {[1, 2, 3].map((label) => (
+          <Step key={label} className={classes.stepperStep}>
+            <StepLabel className={classes.stepperStepLabel} StepIconComponent={ColorlibStepIcon}/>
+          </Step>
+        ))}
```
Comment (ArthurO2002) [inline src/pages/events/createEventsPageComponenets/CreateEvetsStepper.tsx:36]: do not write map this way use variable for map on it

**Review by ArthurO2002 — CHANGES_REQUESTED** [review]

File: src/assets/style/globalVariables.ts (hunk 1 above)
Comment (gor017) [reply src/assets/style/globalVariables.ts:34]: done

File: src/assets/styleJs/events/createEvent.ts (hunk 2 above)
Comment (gor017) [reply src/assets/styleJs/events/createEvent.ts:26]: done

File: src/assets/styleJs/events/createEvent.ts (hunk 3 above)
Comment (gor017) [reply src/assets/styleJs/events/createEvent.ts:54]: done

File: src/assets/styleJs/events/createEvent.ts (hunk 4 above)
Comment (gor017) [reply src/assets/styleJs/events/createEvent.ts:84]: done

File: src/assets/styleJs/events/createEvent.ts (hunk 5 above)
Comment (gor017) [reply src/assets/styleJs/events/createEvent.ts:221]: done

File: src/pages/events/createEvents/stepper/ColorlibStepIconRoot.tsx (hunk 7 above)
Comment (gor017) [reply src/pages/events/createEvents/stepper/ColorlibStepIconRoot.tsx:7]: done

File: .eslintrc.js [hunk 11]
```diff
@@ -32,6 +32,8 @@ module.exports = {
     "prettier",
   ],
   rules: {
+    "no-plusplus": 0,
```
Comment (Antonyan7) [inline .eslintrc.js:35]: Don't add this rules @gor017

File: src/App.tsx [hunk 12]
```diff
@@ -28,7 +29,12 @@ const App = () => {
     }
   })
 
-  return <Box className="bg">{userId ? <PrivateRouts /> : <PublicRoutes />}</Box>
+  return (
+    <ContextProvider>
```
Comment (Antonyan7) [inline src/App.tsx:33]: What you mean by name ContextProvider - Your context should have correct name!

Comment (Antonyan7) [reply src/App.tsx:33]: @gor017

File: src/pages/events/createEvents/EventsContext.tsx [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+export const EventContext = React.createContext(initialEventContext)
+
+function eventReducer(state: Event, action: Action) {
+  switch (action.type) {
+    case 'EVENT_TITLE': {
+      return {...state, id: uuidv4(), eventTitle: action.eventTitle}
+    }
+    case 'EVALUATORS': {
```
Comment (Antonyan7) [inline src/pages/events/createEvents/EventsContext.tsx:33]: You Already Have Action....

File: src/pages/events/createEvents/EventsContext.tsx [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+} = {
+  state: initialState,
+  dispatch: () => ({})
+}
+
+export const EventContext = React.createContext(initialEventContext)
+
+function eventReducer(state: Event, action: Action) {
+  switch (action.type) {
+    case 'EVENT_TITLE': {
```
[the remark this answers is not in the document]
Comment (Antonyan7) [reply src/pages/events/createEvents/EventsContext.tsx:30]: Action.EVENT_TITLE

**Review by Antonyan7 — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 12 above)
Comment (gor017) [reply src/App.tsx:33]: done

File: src/pages/events/createEvents/EventsContext.tsx (hunk 6 above)
Comment (gor017) [reply src/pages/events/createEvents/EventsContext.tsx:69]: done
no, it isn't necessary

File: src/pages/events/createEventsPageComponenets/CreateEvetsStepper.tsx (hunk 9 above)
Comment (gor017) [reply src/pages/events/createEventsPageComponenets/CreateEvetsStepper.tsx:17]: it gives MUI as styled component, it's for making stepper(each values is for stepper's each step rendering)

File: src/pages/events/createEventsPageComponenets/CreateEvetsStepper.tsx (hunk 10 above)
Comment (gor017) [reply src/pages/events/createEventsPageComponenets/CreateEvetsStepper.tsx:36]: done

File: .eslintrc.js [hunk 15]
```diff
@@ -32,6 +32,8 @@ module.exports = {
...
+    "no-console": 0,
     "disable-next-line": [0, { extensions: [".js", ".jsx", ".ts", ".tsx"] }],
```
Comment (ArthurO2002) [inline .eslintrc.js:37]: remove this line

File: .eslintrc.js (hunk 11 above)
Comment (ArthurO2002) [reply .eslintrc.js:35]: I have added no-plusplus rule

File: src/App.tsx (hunk 12 above)
Comment (ArthurO2002) [reply src/App.tsx:33]: Not fixed

File: src/assets/style/globalVariables.ts (hunk 1 above)
Comment (ArthurO2002) [reply src/assets/style/globalVariables.ts:34]: Not fixed

File: src/assets/styleJs/events/createEvent.ts (hunk 2 above)
Comment (ArthurO2002) [reply src/assets/styleJs/events/createEvent.ts:26]: not fixed?

File: src/assets/styleJs/events/createEvent.ts (hunk 3 above)
Comment (ArthurO2002) [reply src/assets/styleJs/events/createEvent.ts:54]: not fixed

File: src/assets/styleJs/events/createEvent.ts (hunk 4 above)
Comment (ArthurO2002) [reply src/assets/styleJs/events/createEvent.ts:84]: not fixed

File: src/assets/styleJs/events/createEvent.ts (hunk 5 above)
Comment (ArthurO2002) [reply src/assets/styleJs/events/createEvent.ts:221]: not fixed

File: src/pages/events/createEvents/EventsContext.tsx [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+  const value = React.useMemo(() => ({
+    state, dispatch
+  }), []);
+
+  return <EventContext.Provider value={value}>{children}</EventContext.Provider>
+}
+
+
+
+
```
Comment (ArthurO2002) [inline src/pages/events/createEvents/EventsContext.tsx:75]: configure eslint and prettier to not have thnigs like this

File: src/pages/events/createEvents/stepper/ColorlibStepIconRoot.tsx (hunk 7 above)
Comment (ArthurO2002) [reply src/pages/events/createEvents/stepper/ColorlibStepIconRoot.tsx:7]: not fixed

File: src/pages/events/createEvents/stepper/ColorlibStepIconRoot.tsx (hunk 8 above)
Comment (ArthurO2002) [reply src/pages/events/createEvents/stepper/ColorlibStepIconRoot.tsx:21]: not fixed

File: src/pages/events/createEventsPageComponenets/CreateEvetsStepper.tsx (hunk 9 above)
Comment (ArthurO2002) [reply src/pages/events/createEventsPageComponenets/CreateEvetsStepper.tsx:17]: should keys be also 1 2 3?

File: src/pages/events/createEventsPageComponenets/CreateEvetsStepper.tsx (hunk 10 above)
Comment (ArthurO2002) [reply src/pages/events/createEventsPageComponenets/CreateEvetsStepper.tsx:36]: not fixed

**Review by ArthurO2002 — CHANGES_REQUESTED** [review]
Not fixed yet

please write done under comments after commit

File: .eslintrc.js (hunk 15 above)
Comment (gor017) [reply .eslintrc.js:37]: done

File: src/pages/events/createEventsPageComponenets/CreateEvetsStepper.tsx (hunk 9 above)
Comment (gor017) [reply src/pages/events/createEventsPageComponenets/CreateEvetsStepper.tsx:17]: yes, there can be also words

File: src/pages/events/createEvents/EventsContext.tsx (hunk 16 above)
Comment (gor017) [reply src/pages/events/createEvents/EventsContext.tsx]: done

File: src/pages/events/createEvents/CreateEvent.tsx [hunk 17]
```diff
@@ -0,0 +1,473 @@
...
+
+  const onChangeEvaluateeSalary = (id: string, value: string) => {
+    const newArray = evaluateesList.map((item: IEvaluatee) => {
+      if (item.id === id) {
+        return {...item, salaryValue: value}
+      }
+      return item
+    })
+    setEvaluateesList(newArray)
+  }
```
Comment (ArthurO2002) [inline src/pages/events/createEvents/CreateEvent.tsx:152]: You can use only 1 function for all of this, it's just taking place

File: src/pages/events/createEvents/CreateEvent.tsx [hunk 18]
```diff
@@ -0,0 +1,473 @@
...
+        {evaluatorCountError ? (
+          <Typography className={classes.evaluatorNumberErrorText}>
+            You need to fill number of evaluators.
+          </Typography>
+        ) : null}
+        <Box className={classes.checkboxManagerBox}>
+          <Checkbox
+            style={{
+              color: '#00A3FF',
+            }}
```
Comment (ArthurO2002) [inline src/pages/events/createEvents/CreateEvent.tsx:295]: Do not write inline styles

File: src/pages/events/createEvents/CreateEvent.tsx [hunk 19]
```diff
@@ -0,0 +1,473 @@
...
+                  InputLabelProps={{style: {fontSize: 14}}}
+                  className={`${classes.evaluatorInput} textField-remove-border`}
+                  label="Number of evaluatees"
+                  variant="outlined"
+                  error={evaluateeCountError}
+                  type="number"
+                  fullWidth
+                  size="small"
+                  onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+                    setEvaluateesCount(parseInt(e.target.value, 10))
```
Comment (ArthurO2002) [inline src/pages/events/createEvents/CreateEvent.tsx:374]: Throw error when the number is with a point

File: src/pages/events/createEvents/CreateEventCriteria.tsx [hunk 20]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import {Typography} from "@mui/material";
+
+ const CreateEventCriteria = () =>
+  (
+    <div>
+      <p>ajskhglaksjglkasjgklasjgl</p>
```
Comment (ArthurO2002) [inline src/pages/events/createEvents/CreateEventCriteria.tsx:7]: What is this?

File: src/pages/events/createEvents/stepper/ColorlibConnector.tsx [hunk 21]
```diff
@@ -0,0 +1,28 @@
...
+    [`& .${stepConnectorClasses.alternativeLabel}`]: {
+      top: 22,
+    },
+    [`&.${stepConnectorClasses.active}`]: {
+      [`& .${stepConnectorClasses.line}`]: {
+        backgroundColor: color.tableBorderColor,
+      },
+    },
+    [`&.${stepConnectorClasses.completed}`]: {
+      [`& .${stepConnectorClasses.line}`]: {
```
Comment (ArthurO2002) [inline src/pages/events/createEvents/stepper/ColorlibConnector.tsx:17]: What is this?
It's not for fixing just asking)

File: src/types.ts [hunk 22]
```diff
@@ -0,0 +1,13 @@
+enum createEventReducerTypes {
+  eventTitle = 'EVENT_TITLE',
+  evaluators = 'EVALUATORS',
+  evaluatees = 'EVALUATEES',
+  criterias = 'CRITERIAS',
```
Comment (ArthurO2002) [inline src/types.ts:5]: Wait to @yavetisyan PR to be fixed or work with him to fix it then migrate this part to there

**Review by ArthurO2002 — CHANGES_REQUESTED** [review]

File: src/pages/events/createEvents/CreateEventCriteria.tsx (hunk 20 above)
Comment (gor017) [reply src/pages/events/createEvents/CreateEventCriteria.tsx:7]: it's just for testing

File: src/pages/events/createEvents/CreateEvent.tsx (hunk 17 above)
Comment (gor017) [reply src/pages/events/createEvents/CreateEvent.tsx]: done

File: src/pages/events/createEvents/CreateEvent.tsx (hunk 18 above)
Comment (gor017) [reply src/pages/events/createEvents/CreateEvent.tsx]: done

File: src/pages/events/createEvents/CreateEvent.tsx (hunk 19 above)
Comment (gor017) [reply src/pages/events/createEvents/CreateEvent.tsx]: it's already throws error

File: src/pages/events/createEvents/CreateEventFirstPage.tsx [hunk 23]
```diff
@@ -0,0 +1,412 @@
...
+
+  const onChangeEvaluateeData = (id: string, value: string, changingValue: string) => {
+    const newArray = evaluateesList.map((item: IEvaluatee) => {
+      if (item.id === id) {
+        return {...item, [changingValue]: value}
+      }
+      return item
+    })
+    setEvaluateesList(newArray)
+  }
```
Comment (ArthurO2002) [inline src/pages/events/createEvents/CreateEventFirstPage.tsx:103]: almost the same function

**Review by ArthurO2002 — CHANGES_REQUESTED** [review]

File: src/pages/events/createEvents/CreateEventFirstPage.tsx [hunk 24]
```diff
@@ -0,0 +1,412 @@
...
+
+  const handleRemoveEvaluatee = (item: IEvaluatee) => {
+    const newList = [...evaluateesList]
+    const index = evaluateesList.indexOf(item)
+    if (index > -1) {
+      newList.splice(index, 1)
+    }
+    setEvaluateesList(newList)
+  }
```
[the remark this answers is not in the document]
Comment (gor017) [reply src/pages/events/createEvents/CreateEventFirstPage.tsx:179]: They are different fields, and two functions are needed

[the remark this answers is not in the document]
Comment (gor017) [reply src/pages/events/createEvents/CreateEventFirstPage.tsx:179]: They are different fields, and two functions are needed

File: src/pages/events/createEvents/CreateEventFirstPage.tsx [hunk 25]
```diff
@@ -0,0 +1,412 @@
...
+    setEvaluatorCountError(false)
+    renderEvaluators(evaluatorsCount)
+  }
+
+  const handleEvaluateesSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault()
+    setEvaluateeCountError(false)
+    renderEvaluatees(evaluateesCount)
+  }
+
```
[the remark this answers is not in the document]
Comment (gor017) [reply src/pages/events/createEvents/CreateEventFirstPage.tsx:115]: They are different fields, and two functions are needed

[the remark this answers is not in the document]
Comment (gor017) [reply src/pages/events/createEvents/CreateEventFirstPage.tsx:115]: They are different fields, and two functions are needed

File: src/pages/events/createEvents/CreateEventFirstPage.tsx (hunk 23 above)
Comment (gor017) [reply src/pages/events/createEvents/CreateEventFirstPage.tsx:102]: They are different fields, and two functions are needed

File: src/pages/events/createEvents/CreateEventFirstPage.tsx [hunk 26]
```diff
@@ -0,0 +1,412 @@
...
+import {IEvaluatee, IEvaluator, ITopManagerEmptyObjInfoType} from './TypesEvents'
+import DeleteIcon from '../../../assets/images/Icons/DeleteIcon'
+import useCreateEventStyles from '../../../assets/styleJs/events/createEvent'
+import {EventContext} from './EventsContext'
+import {v4 as uuidv4} from 'uuid'
+import createEventReducerTypes from '../../../types/createEventTypes'
+
+const CreateEventFirstPage = () => {
+  const classes = useCreateEventStyles()
+  // const navigate = useNavigate()
```
[the remark this answers is not in the document]
Comment (gor017) [reply src/pages/events/createEvents/CreateEventFirstPage.tsx]: done

[the remark this answers is not in the document]
Comment (gor017) [reply src/pages/events/createEvents/CreateEventFirstPage.tsx]: done

**Comment by ArthurO2002** [comment]
Conflicts

File: src/pages/events/createEvents/CreateEventCriteria.tsx [hunk 27]
```diff
@@ -0,0 +1,31 @@
...
+const CreateEventCriteria = () => {
+  const classes = useCreateEventStyles()
+
+  return (
+    <Box>
+      <Box className={classes.infoRootBoxCriteria}>
+        <Box className={classes.criteriaHeaderBox}>
+          <Typography className={classes.criteriaHeaderText}>
+            {' '}
+            Choose Evaluation Criteria{' '}
```
Comment (ArthurO2002) [inline src/pages/events/createEvents/CreateEventCriteria.tsx:15]: What is point of this, use padding if needed

File: src/pages/events/createEvents/CriteriasPaper.tsx [hunk 28]
```diff
@@ -0,0 +1,84 @@
+import React, {useState} from 'react'
+import useCreateEventStyles from '../../../assets/styleJs/events/createEvent'
+import {Box, Button, Checkbox, Grid, Typography} from '@mui/material'
+import {ICriteriaPaperInfo} from './TypesEvents'
+
+const CriteriasPapers = () => {
+  const classes = useCreateEventStyles()
+  // const navigate = useNavigate()
+  // const UseEventContext = () => useContext(EventContext)
```
Comment (ArthurO2002) [inline src/pages/events/createEvents/CriteriasPaper.tsx:9]: remove

**Review by ArthurO2002 — CHANGES_REQUESTED** [review]

File: src/pages/events/createEvents/CreateEventCriteria.tsx (hunk 27 above)
Comment (gor017) [reply src/pages/events/createEvents/CreateEventCriteria.tsx]: done

File: src/pages/events/createEvents/CriteriasPaper.tsx (hunk 28 above)
Comment (gor017) [reply src/pages/events/createEvents/CriteriasPaper.tsx]: done

**Review by ArthurO2002 — APPROVED** [review]

**Pull request merged**